transforms/universal/ededup/python/src/ededup_transform_base.py

```diff
@@ -0,0 +1,154 @@
+# (C) Copyright IBM Corp. 2024.
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+from argparse import ArgumentParser, Namespace
+from typing import Any
+
+import pyarrow as pa
+from data_processing.transform import AbstractTransform
+from data_processing.transform import AbstractTableTransform, TransformConfiguration
+from data_processing.utils import GB, CLIArgumentProvider, TransformUtils
+
+
+REQUEST_LEN = 8192
+short_name = "ededup"
+cli_prefix = f"{short_name}_"
+
+
+class HashFilter:
+    """
+    Implements hash
+    """
+
+    def __init__(self, params: dict[str, Any]):
+        """
+        initialize set of local hashes
+        """
+        self.hashes = set()
+
+    def get_unique(self, ha: list[str]) -> list[str]:
+        """
+        Get list of unique hashes
+        :param ha: new set of hashes
+        :return: list of unique ones
+        """
+        unique = []
+        for h in ha:
+            if h not in self.hashes:
+                # If a hash does not exist, add it to unique and add to the local set
+                self.hashes.add(h)
+                unique.append(h)
+        return unique
+
+    def get_hash_size(self) -> tuple[int, float]:
+        """
+        Get size of created hashes for statistics
+        :return: size of the local set and its memory footprint
+        """
+        return len(self.hashes), TransformUtils.deep_get_size(self.hashes) / GB
+
+
+class EdedupTransformBase(AbstractTableTransform):
+    """
+    Implements dedup table transformer.
+    """
+
+    def __init__(self, config: dict):
+        """
+        Initialize based on the dictionary of configuration information.
+        The dictionary should contain the following:
+            doc_column - name of the doc column
+        """
+        super().__init__(config)
+        self.doc_column = config.get("doc_column", "contents")
```

Comment:
would be better to define global constants for these and other strings, similar to the way many other transforms do it.

Reply:
its a matter of preference